62.CAD/fragment_shader.hlsl

```diff
@@ -15,6 +16,68 @@ void endInvocationInterlockEXT();
 // Write a general one, and maybe another one that uses precomputed values, and move these to somewhere nice in our builtin hlsl shaders if we don't have one
 // See: https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
 
+template<typename float_t, typename CurveType>
+struct LineStyleClipper
+{
+    using float2_t = vector<float_t, 2>;
+
+    static LineStyleClipper<float_t, CurveType> construct(uint32_t styleIdx, 
+                                                          CurveType curve,
+                                                          typename CurveType::ArcLenCalculator arcLenCalc,
+                                                          float totalArcLen)
+    {
+        LineStyleClipper<float_t, CurveType> ret = { styleIdx, curve, arcLenCalc, totalArcLen };
+        return ret;
+    }
+    
+    struct ArrayAccessor
+    {
+        uint32_t styleIdx;
+        using value_type = float;
+
+        float operator[](const uint32_t ix)
+        {
+            return lineStyles[styleIdx].stipplePattern[ix];
+        }
+    };
+    
+    float2_t operator()(const float t)
+    {
+        const float arcLen = arcLenCalc.calcArcLen(t);
+        float_t tMappedToPattern = frac(arcLen / float(globals.screenToWorldRatio) * lineStyles[styleIdx].recpiprocalStipplePatternLen + lineStyles[styleIdx].phaseShift);
+        ArrayAccessor stippleAccessor = { styleIdx };
+        uint32_t patternIdx = nbl::hlsl::upper_bound(stippleAccessor, 0, lineStyles[styleIdx].stipplePatternSize, tMappedToPattern);
+        
+        if(patternIdx & 0x1)

```

Comment:
Comment that odd means a no draw section and current candidate should split into two nearest draw sections

Reply:
done